src/classic/clvm_tools/stages/stage_2/module.rs

```diff
@@ -595,56 +603,61 @@ fn build_macro_lookup_program(
     }
 }
 
+#[allow(clippy::too_many_arguments)]
 fn add_one_function(
     allocator: &mut Allocator,
     args_root_node: &NodePath,
     macro_lookup_program: NodePtr,
     constants_symbol_table: &[(NodePtr, Vec<u8>)],
-    compiled_functions_: HashMap<Vec<u8>, NodePtr>,
+    // Note: mut here means: the body of this function will mutate this by-value
+    // parameter, not that the mutability is visible to the caller.
+    //
+    // My own style generally avoid this, but reviewers tend to dislike having a
+    // phantom name that is copied, so i am using the more brief style here.
+    mut compiled: CompileOutput,
     name: &[u8],
     lambda_expression: NodePtr,
-) -> Result<HashMap<Vec<u8>, NodePtr>, EvalErr> {
-    let mut compiled_functions = compiled_functions_;
-    m! {
-        com_atom <- allocator.new_atom("com".as_bytes());
-        opt_atom <- allocator.new_atom("opt".as_bytes());
+    has_constants_tree: bool,
+) -> Result<CompileOutput, EvalErr> {
+    let com_atom = allocator.new_atom("com".as_bytes())?;
+    let opt_atom = allocator.new_atom("opt".as_bytes())?;
 
-        le_first <- first(allocator, lambda_expression);
-        local_symbol_table <- symbol_table_for_tree(
-            allocator, le_first, args_root_node
-        );
-        let mut all_symbols = local_symbol_table;
-        let _ = all_symbols.append(&mut constants_symbol_table.to_owned());
-        lambda_form_content <- rest(allocator, lambda_expression);
-        lambda_body <- first(allocator, lambda_form_content);
-        quoted_lambda_expr <- quote(allocator, lambda_body);
-        all_symbols_list_sexp <-
-            map_m(
-                allocator,
-                &mut all_symbols.iter(),
-                &|allocator, pair| m! {
-                    path_atom <- allocator.new_atom(&pair.1);
-                    enlist(allocator, &[pair.0, path_atom])
-                }
-            );
+    let function_args = first(allocator, lambda_expression)?;
+    let local_symbol_table = symbol_table_for_tree(allocator, function_args, args_root_node)?;
+    let mut all_symbols = local_symbol_table;
```

Comment:
instead of declaring a new variable `all_symbols`, you could just make `local_symbol_table` mutable.
This is old code though, so not really related to this PR